From 9eae74e97bb0425f3b106ee096363549e2c17865 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Sun, 14 Jul 2013 13:15:03 +0300
Subject: [PATCH 0818/1825] alp: fix Revision ID read to use MV_U32 instead of
 MV_U8

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit 3fcbd6ffba69df611a557a17d05897130fa506d6

	- Fixed read of revision ID to read U32 instead of U8
	  (The revision is bits 8-11 - always read 0x0 with U8)

Change-Id: I5518fbeea30fdfd1036e8b9cb536f36cb80862a3
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2609
Reviewed-by: Kosta Zertsekel <konszert@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c        |    4 ++--
 .../avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h        |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)
 mode change 100644 => 100755 arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
index d40b0cf..97a2ff2 100755
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.c
@@ -1125,9 +1125,9 @@ MV_U16 mvCtrlModelGet(MV_VOID)
 *       8bit desscribing Marvell controller revision number
 *
 *******************************************************************************/
-MV_U8 mvCtrlRevGet(MV_VOID)
+MV_U32 mvCtrlRevGet(MV_VOID)
 {
-	MV_U8 value;
+	MV_U32 value;
 
 	value = MV_REG_READ(DEV_VERSION_ID_REG);
 	return  ((value & (REVISON_ID_MASK) ) >> REVISON_ID_OFFS);
diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
old mode 100644
new mode 100755
index bfebc58..34cc3c4
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/mvCtrlEnvLib.h
@@ -316,7 +316,7 @@ MV_TDM_UNIT_TYPE mvCtrlTdmUnitTypeGet(MV_VOID);
 MV_U32 mvCtrlTdmUnitIrqGet(MV_VOID);
 MV_U32 mvCtrlDevFamilyIdGet(MV_U16 ctrlModel);
 MV_U16 mvCtrlModelGet(MV_VOID);
-MV_U8 mvCtrlRevGet(MV_VOID);
+MV_U32 mvCtrlRevGet(MV_VOID);
 MV_STATUS mvCtrlNameGet(char *pNameBuff);
 MV_U32 mvCtrlModelRevGet(MV_VOID);
 MV_STATUS mvCtrlModelRevNameGet(char *pNameBuff);
-- 
1.7.5.4

